### PR TITLE
OCPBUGS-24977: Updating cluster-monitoring-operator-container image to be consistent with ART

### PR DESCRIPTION
**Pull request by simonpasquier**
Supersedes #2207

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/openshift/cluster-monitoring-operator
 
-go 1.20
+go 1.21
```
Comment (machine424) [inline go.mod:3]: Maybe we could automate the upgrade of this as well.

**Comment by simonpasquier** [comment]
/close
https://github.com/openshift/cluster-monitoring-operator/pull/2211 is the one we want to merge.

**Pull request closed without merging**